python/test/unit/language/test_core.py

```diff
@@ -345,6 +345,12 @@ def _mod_operation_ill_conditioned(dtype_x, dtype_y) -> bool:
     ]
 
 
+def test_dtype_codegen():
+    assert repr(triton.language.float8e4b15x4) == "triton.language.float8e4b15x4"
+    assert repr(triton.language.float16) == "triton.language.float16"
+    assert repr(triton.language.bfloat16) == "triton.language.bfloat16"
```

Comment:
Is it worth checking all of the dtypes?  (i.e. iterating over dtype.SINT_TYPES, UINT_TYPES, etc?)